src/resource.jl

```diff
@@ -244,13 +244,21 @@ function download(server::String, resource::String, path::String)
     end
 end
 
-function serve_file(http::HTTP.Stream, path::String)
+function serve_file(http::HTTP.Stream, path::String, chunksize=2*1024*1024)
     HTTP.setheader(http, "Content-Length" => string(filesize(path)))
     # We assume that everything we send is gzip-compressed (since they're all tarballs)
     HTTP.setheader(http, "Content-Encoding" => "gzip")
     startwrite(http)
 
-    # Open the path, write it out directly to the HTTP stream
-    open(io -> write(http, read(io, String)), path)
+    # Open the path, write it out directly to the HTTP stream in chunks
+    open(path, "r") do io
+        while !eof(io)
+            # We set `all` to `false` so that we aren't artificially slowing down time to first byte;
+            # if the disk is a little busy, we're happy to start transmitting a smaller chunk of data,
+            # as it is very likely by the time the first small chunk is done transmitting we'll be ready
+            # with a new chunk of data from disk, as the disk is, in general, faster than the network.
+            write(http, read(io, chunksize, all=false))
```

Comment:
I wonder if it's better to allocate a buffer and reuse it. cc @Keno

Reply:
Yes, that would be better, unless you plan to do something fancy with parallel streams.

Reply:
Death to all allocations.